File: Bike_Rent/bike.py
## your code goes here ##
class BikeRental:
    def __init__(self, stock=100):
        self.stock = stock
        print("Welcome to rental bike shop.")

    def displaystock(self):
        print(f"Total number of stock available: {self.stock}")

bike_shop = BikeRental()

class Customer(BikeRental):
  bill = 0
  def __init__(self,number_of_bikes, rentalBasis,number_of_days_or_weeks):
    self.number_of_bikes = number_of_bikes
    self.rentalBasis = rentalBasis.lower()
    self.number_of_days_or_weeks = number_of_days_or_weeks

  def rentBike(self):
    if self.number_of_bikes <= 0:
      print("Number of bikes should be positive!")
    elif self.number_of_bikes > bike_shop.stock:
      print(f" Only {bike_shop.stock} available to rent.")
    else:
      print(f"Number of bikes rented: {self.number_of_bikes}")
      bike_shop.stock -= self.number_of_bikes
      print("Available bikes",bike_shop.stock)

  def returnBike(self):
     if self.rentalBasis == 'day':
      rate = 100
     elif self.rentalBasis == 'week':
      rate = 500
     else:
       print("Invalid rental basis.")
       return
     total = self.number_of_days_or_weeks * rate * self.number_of_bikes

     if 3<= self.number_of_bikes <= 5:
      discount = total * 0.3
      total -= discount
      print("You are eligible for Family rental promotion of 30% discount")

     print(f"Total bill: ₹{total}")
     Customer.bill = total

File: Bike_Rent/bike.js
// Initial stock of bikes
let stock = 100;

// DOM Elements
const stockCount = document.getElementById('stock-count');
const rentalForm = document.getElementById('rental-form');
const returnForm = document.getElementById('return-form');
const billDiv = document.getElementById('bill');

// Rent a bike
rentalForm.addEventListener('submit', function (event) {
    event.preventDefault();

    const numberOfBikes = parseInt(document.getElementById('number-of-bikes').value);
    const rentalBasis = document.getElementById('rental-basis').value.toLowerCase();
    const daysOrWeeks = parseInt(document.getElementById('days-or-weeks').value);

    if (numberOfBikes <= 0) {
        alert("Number of bikes should be positive!");
        return;
    }

    if (numberOfBikes > stock) {
        alert(`Only ${stock} bikes are available to rent.`);
        return;
    }

    // Rent bikes
    stock -= numberOfBikes;
    stockCount.textContent = stock;

    let rate = rentalBasis === 'day' ? 100 : rentalBasis === 'week' ? 500 : 0;
    if (rate === 0) {
        alert("Invalid rental basis.");
        return;
    }

    let total = rate * daysOrWeeks * numberOfBikes;

    // Apply discount for 3-5 bikes
    if (numberOfBikes >= 3 && numberOfBikes <= 5) {
        let discount = total * 0.3;
        total -= discount;
        alert("You are eligible for a 30% discount.");
    }

    billDiv.innerHTML = `Total Bill: ₹${total}`;
});

// Return bikes
returnForm.addEventListener('submit', function (event) {
    event.preventDefault();

    const numberOfBikesReturned = parseInt(document.getElementById('number-of-bikes').value);

    if (numberOfBikesReturned <= 0) {
        alert("Invalid number of bikes.");
        return;
    }

    // Return bikes and update stock
    stock += numberOfBikesReturned;
    stockCount.textContent = stock;

    alert(`You have returned ${numberOfBikesReturned} bikes.`);
    billDiv.innerHTML = ''; // Clear the bill after return
});
